src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1753,6 +1753,20 @@ private void doFileSelectionModeChanged(PropertyChangeEvent e) {
     private void doMultiSelectionChanged(PropertyChangeEvent e) {
         if (getFileChooser().isMultiSelectionEnabled()) {
             listSelectionModel.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+
+            //Check if any files are selected before setting multi-Selection.
+            //If selected, retain them and update the selected files in FileChooser class.
+            File[] selectedFiles = null;
+            if (getFileChooser().getSelectedFiles().length != 0) {
+                selectedFiles = getFileChooser().getSelectedFiles();
+            } else if (getFileChooser().getSelectedFile() != null) {
+                selectedFiles = new File[1];
+                selectedFiles[0] = getFileChooser().getSelectedFile();
+            }
+
```

Comment:
Is this problem seen only when single file is selected? What happens when more than one file is selected?
By default, is MultiSelection enabled on JfileChooser?

In FilePane.java, i see that get/setSelectedFiles() is exclusively used when multi-selection mode is enabled and get/setSelectedFile() is used when multi-selection is not enabled. Using getSelectedFile() to change the state when multi-selection is enabled might introduce state corruption issues in other scenarios.

Reply:
This problem occurs when user switch the mode from single selection to multi selection, that too on specified scenario. By default multiselection will be disabled. When user selects a file in single selection and then switch to multi selection the selected single file is not retained in selectedFiles (Even though its a single file). So In order to handle this situation, whenever user switch from any state to multiselection enabled state, the previously selected files(if present) is set to selectedFiles().

Reply:
Other state are not altered or modified during the process, just the selected files are set to setSelectedFiles.